### PR TITLE
Enforce include ordering via clang-format

### PR DESCRIPTION
**Pull request by msmk0**
Update the `clang-format` configuration to enforce the grouping and ordering of includes. Includes are always ordered into the following groups separated by an empty line in the listed order.

1. (Only for `.cpp` files) The corresponding main header file.
2. All Acts headers. These come first, to avoid having symbols magically available, e.g. from standard library, that are not explicitly included.
3. Standard library includes.
4. Other system includes.
5. Everything else, including local headers. 

The updated configuration is applied to all files.

### CODE REVIEW

File: Core/include/Acts/Propagator/Propagator.hpp [hunk 1]
```diff
@@ -30,6 +21,14 @@
 #include "Acts/Utilities/Definitions.hpp"
 #include "Acts/Utilities/Result.hpp"
 #include "Acts/Utilities/Units.hpp"
+#include "Acts/Utilities/detail/ReferenceWrapperAnyCompat.hpp"
```
Comment (paulgessinger) [inline Core/include/Acts/Propagator/Propagator.hpp:24]: This would still need to go up top.

**Comment by paulgessinger** [comment]
Probably a good idea. We need to think about when to merge this, though, because it will cause merge conflicts everywhere.

File: Core/include/Acts/Propagator/Propagator.hpp (hunk 1 above)
Comment (msmk0) [reply Core/include/Acts/Propagator/Propagator.hpp]: Fixed for the `clang-format` 10 that I am using locally. This does not work with the `clang-format` version available on the CI. Should we bump it?

Comment (paulgessinger) [reply Core/include/Acts/Propagator/Propagator.hpp]: We could. But you could probably also wrap this one in comments disabling the formatting.

**Comment by msmk0** [comment]
> Probably a good idea. We need to think about when to merge this, though, because it will cause merge conflicts everywhere.

My suggestion is to merge it all at once when it is ready. The CI now shows a few include-what-you-use violations that also need to be fixed.

**Comment by paulgessinger** [comment]
How are we looking on this one?

**Comment by msmk0** [comment]
I just added the fixes to the missing include files. It turned out that these are not that many after all. I think we can leave them as part of this PR.

**Comment by paulgessinger** [comment]
Ok let me approve this then so we can merge this once it goes green.

File: Tests/UnitTests/Plugins/Cuda/Seeding/SeedfinderCudaTest.cpp [hunk 2]
```diff
@@ -6,15 +6,7 @@
...
-#include <chrono>
-#include <fstream>
-#include <iomanip>
-#include <iostream>
-#include <sstream>
-#include <utility>
-
-#include <boost/type_erasure/any_cast.hpp>
-
+#include "Acts/Plugins/Cuda/Seeding/Seedfinder.hpp"
```
Comment (beomki-yeo) [inline Tests/UnitTests/Plugins/Cuda/Seeding/SeedfinderCudaTest.cpp:9]: This file is not compiled after this line was moved to the top. Looks like the dependency is broken ?

```
In file included from /home/bomki/Projects/ACTS/Tests/UnitTests/Plugins/Cuda/Seeding/SeedfinderCudaTest.cpp:9:
/home/bomki/Projects/ACTS/Plugins/Cuda/include/Acts/Plugins/Cuda/Seeding/Seedfinder.hpp:30:7: error: ‘Seedfinder’ is not a class template
 class Seedfinder<external_spacepoint_t, Acts::Cuda> {
       ^~~~~~~~~~
```

Comment (paulgessinger) [reply Tests/UnitTests/Plugins/Cuda/Seeding/SeedfinderCudaTest.cpp:9]: Thanks for reporting! This is exactly what we need the CUDA CI job for.

**Pull request merged**